query/src/main/scala/filodb/query/exec/rangefn/MiscellaneousFunction.scala

```diff
@@ -78,3 +79,56 @@ case class LabelReplaceFunction(funcParams: Seq[Any])
     return rangeVectorKey;
   }
 }
+
+case class LabelJoinFunction(funcParams: Seq[Any])
+  extends MiscellaneousFunction {
+
+  val labelIdentifier: String = "[a-zA-Z_][a-zA-Z0-9_:\\-\\.]*"
+
+  require(funcParams.size >= 2,
+    "expected at least 3 argument(s) in call to label_join")
+
+  val dstLabel: String = funcParams(0).asInstanceOf[String]
+  val separator: String = funcParams(1).asInstanceOf[String]
+
+  require(dstLabel.asInstanceOf[String].matches(labelIdentifier), "Invalid destination label name in label_join()")
+  var srcLabel = ArrayBuffer[String]()
+  funcParams.drop(2).foreach(x => {
+    require(x.asInstanceOf[String].matches(labelIdentifier),
+      "Invalid source label name in label_join()")
+    srcLabel += x.asInstanceOf[String]
+  })
+
+  override def execute(source: Observable[RangeVector]): Observable[RangeVector] = {
+    source.map { rv =>
+      val newLabel = labelJoinImpl(rv.key)
+      IteratorBackedRangeVector(newLabel, rv.rows)
+    }
+  }
+
+  def labelJoinImpl(rangeVectorKey: RangeVectorKey): RangeVectorKey = {
+    var srcLabelValues = ArrayBuffer[String]()
+
+    srcLabel.
+      foreach(x => if (rangeVectorKey.labelValues.contains(ZeroCopyUTF8String(x)))
+        srcLabelValues += rangeVectorKey.labelValues.get(ZeroCopyUTF8String(x)).get.toString
+      else
+        srcLabelValues += ""
+      )
+
+    val labelJoinValue = srcLabelValues.mkString(separator)
```

Comment:
This statement can go inside else. Change if condition to length ==0

Reply:
Even if srcLabelValues has length 0, the new label value should have separator. So will have to check for labelJoinValue length